plugins-builtin/src/main/scala/za/co/absa/enceladus/plugins/builtin/errorsender/mq/KafkaErrorSenderPluginImpl.scala

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.plugins.builtin.errorsender.mq
+
+import org.apache.log4j.LogManager
+import org.apache.spark.sql.functions.{col, explode, lit, size, struct}
+import org.apache.spark.sql.types.DataTypes
+import org.apache.spark.sql.{DataFrame, Encoders}
+import za.co.absa.enceladus.plugins.api.postprocessor.PostProcessor
+import za.co.absa.enceladus.plugins.builtin.common.mq.kafka.KafkaConnectionParams
+import za.co.absa.enceladus.plugins.builtin.errorsender.DceError
+import za.co.absa.enceladus.plugins.builtin.errorsender.mq.KafkaErrorSenderPluginImpl.SingleErrorStardardized
+import za.co.absa.enceladus.utils.schema.SchemaUtils
+import KafkaErrorSenderPluginImpl._
+import za.co.absa.enceladus.plugins.builtin.errorsender.mq.kafka.KafkaErrorSenderPlugin
+import za.co.absa.enceladus.plugins.builtin.errorsender.params.ErrorSenderPluginParams
+import za.co.absa.enceladus.plugins.builtin.errorsender.params.ErrorSenderPluginParams.ErrorSourceId
+import za.co.absa.enceladus.utils.error.ErrorMessage.ErrorCodes
+
+import scala.util.{Failure, Success, Try}
+
+
+case class KafkaErrorSenderPluginImpl(connectionParams: KafkaConnectionParams,
+                                      keySchemaRegistryConfig: Map[String, String],
+                                      valueSchemaRegistryConfig: Map[String, String]) extends PostProcessor {
+
+  private val log = LogManager.getLogger(classOf[KafkaErrorSenderPluginImpl])
+
+  override def close(): Unit = {}
+
+  /**
+   * When data is ready, the error record(s) are pusblished to kafka.
+   *
+   * @param dataFrame error data only.
+   * @param params    Additional key/value parameters provided by Enceladus.
+   */
+  override def onDataReady(dataFrame: DataFrame, params: Map[String, String]): Unit = {
+    if (!SchemaUtils.fieldExists(ColumnNames.enceladusRecordId, dataFrame.schema)) {
+      throw new IllegalStateException(
+        s"${this.getClass.getName} requires ${ColumnNames.enceladusRecordId} column to be present in the dataframe!"
+      )
+    }
+
+    val errorSenderParams = Try(ErrorSenderPluginParams.fromMap(params)) match {
+      case Success(params) => params
```

Comment:
There are too many things called params, while not always meaning the same, right?

Reply:
True, fixed.